types/mui-datatables/index.d.ts

```diff
@@ -159,120 +171,132 @@ export interface MUIDataTableIsRowCheck {
+        selectedRows: { data: Array<{ index: number; dataIndex: number }>; lookup: { [key: number]: boolean } },
         displayData: Array<{ data: any[]; dataIndex: number }>,
         setSelectedRows: (rows: number[]) => void
     ) => React.ReactNode;
-    disableToolbarSelect?: boolean;
-    download?: boolean;
-    downloadOptions?: Partial<{
+    download: boolean;
+    downloadOptions: Partial<{
         filename: string;
         separator: string;
         filterOptions: Partial<{ useDisplayedColumnsOnly: boolean; useDisplayedRowsOnly: boolean }>;
     }>;
-    elevation?: number;
-    expandableRows?: boolean;
-    expandableRowsOnClick?: boolean;
-    filter?: boolean;
-    filterType?: FilterType;
-    fixedHeader?: boolean;
-    fixedHeaderOptions?: {
-        xAxis: boolean;
-        yAxis: boolean;
-    };
-    isRowExpandable?: (dataIndex: number, expandedRows?: MUIDataTableIsRowCheck) => boolean;
-    isRowSelectable?: (dataIndex: number, selectedRows?: MUIDataTableIsRowCheck) => boolean;
-    onCellClick?: (
+    elevation: number;
+    enableNestedDataAccess: string;
+    expandableRows: boolean;
+    expandableRowsHeader: boolean;
+    expandableRowsOnClick: boolean;
+    filter: boolean;
+    filterType: FilterType;
+    fixedHeader: boolean;
+    fixedSelectColumn: boolean;
+    isRowExpandable: (dataIndex: number, expandedRows?: MUIDataTableIsRowCheck) => boolean;
+    isRowSelectable: (dataIndex: number, selectedRows?: MUIDataTableIsRowCheck) => boolean;
+    onCellClick: (
         colData: any,
         cellMeta: { colIndex: number; rowIndex: number; dataIndex: number; event: React.MouseEvent }
     ) => void;
-    onChangePage?: (currentPage: number) => void;
-    onChangeRowsPerPage?: (numberOfRows: number) => void;
-    onColumnSortChange?: (changedColumn: string, direction: string) => void;
-    onColumnViewChange?: (changedColumn: string, action: string) => void;
+    onChangePage: (currentPage: number) => void;
+    onChangeRowsPerPage: (numberOfRows: number) => void;
+    onColumnSortChange: (changedColumn: string, direction: 'asc' | 'desc') => void;
     /**
      * A callback function that triggers when the user downloads the CSV file.
      * In the callback, you can control what is written to the CSV file.
      * Return false to cancel download of file.
      */
-    onDownload?: (
+    onDownload: (
         buildHead: (columns: any) => string,
         buildBody: (data: any) => string,
         columns: any,
-        data: any,
+        data: any
     ) => string | boolean;
-    onFilterChange?: (changedColumn: string, filterList: any[], type: FilterType | 'chip' | 'reset') => void;
-    onFilterDialogOpen?: () => void;
-    onFilterDialogClose?: () => void;
-    onRowClick?: (rowData: string[], rowMeta: { dataIndex: number; rowIndex: number }) => void;
-    onRowsDelete?: (rowsDeleted: {
+    onFilterChange: (changedColumn: string, filterList: any[], type: FilterType | 'chip' | 'reset') => void;
```

Comment:
shuld stay optional IMO:
https://github.com/gregnb/mui-datatables/blob/3ae8302d4c006ee47f951ac0ba113b7fb1b0eba2/src/MUIDataTable.js#L1258
https://github.com/gregnb/mui-datatables/blob/3ae8302d4c006ee47f951ac0ba113b7fb1b0eba2/src/MUIDataTable.js#L195
```suggestion
    onFilterChange?: (changedColumn: string, filterList: any[], type: FilterType | 'chip' | 'reset') => void;
```